src/ring_theory/fractional_ideal.lean

```diff
@@ -166,7 +206,20 @@ mem_one_iff.mpr ⟨x, rfl⟩
 lemma one_mem_one : (1 : P) ∈ (1 : fractional_ideal f) :=
 mem_one_iff.mpr ⟨1, f.to_map.map_one⟩
 
-@[simp] lemma coe_one : ↑(1 : fractional_ideal f) = f.coe_submodule 1 := rfl
+/-- `(1 : fractional_ideal f)` is defined as the R-submodule `f(R) ≤ K`.
+
+However, this is not definitionally equal to `1 : submodule R K`,
+which is proved in the actual `simp` lemma `coe_one`. -/
+lemma coe_one_eq_coe_submodule_one :
+  ↑(1 : fractional_ideal f) = f.coe_submodule (1 : ideal R) :=
+rfl
+
+@[simp] lemma coe_one :
```

Comment:
Possibly `norm_cast`?
```suggestion
@[simp, norm_cast] lemma coe_one :
```

Reply:
Good idea. I'll go through the file later to see if there are some more `@[norm_cast]`s needed.